tests/unit_test/test_curve.py

```diff
@@ -0,0 +1,122 @@
+import numpy as np
+import unittest
+from beprof.curve import Curve
+
+
+class TestCurveInit(unittest.TestCase):
+    """
+    Testing Curve initialization and .x .y
+    """
+    def test_list_init(self):
+        simple_lists = [[-12, 1], [-1, 7], [0, 3], [3, 17]]
+        c = Curve(simple_lists)
+        assert list(c.x) == [e[0] for e in simple_lists]
+        assert list(c.y) == [e[1] for e in simple_lists]
+
+    def test_numpy_array_init(self):
+        numpy_array = np.array([[-12, 1], [-1, 7], [0, 3], [3, 17]])
+        c = Curve(numpy_array)
+        assert np.array_equal(c, numpy_array)
+
+    def test_numpy_view_init(self):
+        # todo
+        pass
+
+    def test_empty_init(self):
+        with self.assertRaises(TypeError):
+            Curve()
+        with self.assertRaises(IndexError):
```

Comment:
Lets postpone this discussion - please create an issue to investigate how beprof behaves with non-numerical data.
Very often it happens that experimental data is a mixture of numbers and plain text. If we read such stuff from file and feed it to beprof classes, then strange things might happen. It would be good to do some filtering on init.